R/zoop_downloader.R

```diff
@@ -843,37 +865,61 @@ Zoopdownloader <- function(
+    if (!file.exists(file.path(Data_folder, "bugsFRP2021.csv")) | Redownload_data) {
+      Tryer(n=3, fun=utils::download.file, url="https://portal.edirepository.org/nis/dataviewer?packageid=edi.269.3&entityid=8de785ee47220f3893654478c79b5f8f",
+            destfile=file.path(Data_folder, "bugsFRP2021.csv"), mode="wb", method=Download_method)
+      Tryer(n=3, fun=utils::download.file, url="https://portal.edirepository.org/nis/dataviewer?packageid=edi.269.3&entityid=fa84750e51c319d309f97357b7d34315",
+            destfile=file.path(Data_folder, "sitesFRP2021.csv"), mode="wb", method=Download_method)
+
     }
 
-    zoo_FRP_Macro <- readr::read_csv(file.path(Data_folder, "bugsFRP2018.csv"),
-                                     col_types = "ccccddddddddccdddcddc", na=c("", "NA"))
+    zoo_FRP_Macro <- readr::read_csv(file.path(Data_folder, "bugsFRP2021.csv"), na=c("", "NA"))
+    sites_FRP_Macro <- readr::read_csv(file.path(Data_folder, "sitesFRP2021.csv"), na=c("", "NA"))
+
+    #join environmental data to taxa counts and fix some wonky names
+    FRP_allmac = dplyr::left_join(dplyr::select(zoo_FRP_Macro, -Date, -Location), sites_FRP_Macro, by = "VisitNo") %>%
+      dplyr::mutate(CommonName = dplyr::case_when(CommonName == "Fish larvae" ~ "Fish UNID", #fix some wonky common names
+                                    CommonName == "Insect Unid" ~ "Insect UNID",
+                                    CommonName == "Calanoid copepod (gravid)" ~ "Calanoid UNID",
+                                    CommonName == "Hymenoptera UNID" ~ "Hymenoptera Other",
+                                    CommonName == "Tricoptera larvae Other" ~ "Tricoptera larvae UNID",
+                                    CommonName == "Palaemonectes" ~ "Palaemon",
+                                    CommonName == "Palaemonetes" ~ "Palaemon",
+                                    CommonName == "Hymenoptera UNID" ~ "Hymenoptera Other",
+                                    CommonName == "Asellidae UNID" ~ "Asellidae",
+                                    CommonName == "Diptera adult" ~ "Diptera Adult",
+                                    CommonName == "Coleoptera other" ~ "Coleoptera Other",
+                                    CommonName == "Tricoptera larvae UNID"~"Trichoptera larvae Other",
+                                    TRUE ~ CommonName))
 
     #Already in long format
-    data.list[["FRP_Macro"]] <- zoo_FRP_Macro%>%
-      dplyr::filter(.data$Sampletype=="trawl")%>%
+    data.list[["FRP_Macro"]] <- FRP_allmac%>%
+      dplyr::filter(.data$GearTypeAbbreviation %in% c("MAC", "MACOBL", "MACBEN"))%>%
       dplyr::mutate(Date=lubridate::parse_date_time(.data$Date, "%m/%d/%Y", tz="America/Los_Angeles"))%>%
-      dplyr::mutate(Station=dplyr::recode(.data$Station, `Lindsey Tules`="Lindsey tules", LinBR="LinBr", MINSLO1="MinSlo1", ProBR="ProBr", WinBR="WinBr"))%>% #Rename inconsistent station names to match
```

Comment:
I assume this means that all the station names in the zoops and bugs tables now match the station names in the station location table?

Reply:
FRP is no longer using station names, it just has 'location', which are broader regions. I changed the integrated dataset to use "Location" instead of "Station" for the station column in the integrated dataset.